lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -395,7 +395,12 @@ def send_request_cgi(opts = {}, timeout = 20, disconnect = true)
     return unless res
 
     if opts['keep_cookies'] && res.headers['Set-Cookie'].present?
-      cookie_jar.parse_and_merge(res.headers['Set-Cookie'])
+      cookies = cookie_jar.parse(res.headers['Set-Cookie'])
+
+      cookies.each do |c|
+        c.origin = "http#{ssl ? 's' : ''}://#{vhost}:#{rport}"
```

Comment:
can the vhost ever be nil? I don't think it can I think it defaults to the ip if it's not set but just wanted to call it out to be sure

Reply:
Aye VHOST can be nil, in which case it gets set to RHOST 👍 
https://github.com/rapid7/metasploit-framework/blob/569660b8904a7a34c75f2f9ea044cda75c64640f/lib/msf/core/exploit/remote/http_client.rb#L653